### DISCUSSION TITLE
How to scope global styles?

### DISCUSSION THREAD

**Discussion by michalHT** [chakra-ui/chakra-ui#3401] [Q&A] [answered]
I'm using Chakra only on selected views in my app and I don't want Chakra global styles to pollute the rest of the app.

How can I scope the global styles applied by Chakra to only have effect within a particular div?

In other words:

- instead of applying to `*` apply to `#chakra-root *`,
- instead of applying to `body` apply to `#chakra-root`,
- instead of applying to `h1` apply to `#chakra-root h1`,
etc.

**Comment by with-heart** [comment, marked as answer]
Hi @michalHT! We don't have a mechanism for doing this automatically, but you could set `resetCSS={false}` on `ChakraProvider`, create your own implementation of [our `CSSReset` component](https://github.com/chakra-ui/chakra-ui/blob/main/packages/css-reset/src/css-reset.tsx) that scopes the styles how you'd prefer, and then render that component.

**Reply by michalHT** [reply]
Awesome :)
Thank you @with-heart!
